### PR TITLE
Refactor jobs

### PR DESCRIPTION
**Pull request by rpitonak**
Tested locally. I will be glad for comments on design.

TODO

- [x] fix tests

### CODE REVIEW

File: packit_service/service/events.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        self, repo_namespace: str, repo_name: str, tag_name: str, https_url: str
+    ):
+        super(ReleaseEvent, self).__init__(JobTriggerType.release)
+        self.repo_namespace = repo_namespace
+        self.repo_name = repo_name
+        self.tag_name = tag_name
+        self.https_url = https_url
+
+    def get_dict(self) -> dict:
+        result = self.__dict__
```
Comment (phracek) [inline packit_service/service/events.py]: why not simply {}

File: packit_service/service/events.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        result["trigger"] = str(result["trigger"])
+        return result
```
Comment (phracek) [inline packit_service/service/events.py]: what about:
```python
return {"trigger" : str(result["trigger"])}
```

File: packit_service/service/events.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        self.ref = ref
+        self.branch = branch
+        self.msg_id = msg_id
+
+    def get_dict(self) -> dict:
+        result = self.__dict__
+        # whole dict have to be JSON serializable because of redis
+        result["trigger"] = str(result["trigger"])
+        result["topic"] = str(result["topic"])
+        return result
```
Comment (phracek) [inline packit_service/service/events.py]: The same as in `ReleaseEvent`

File: packit_service/service/events.py (hunk 1 above)
Comment (TomasTomecek) [reply packit_service/service/events.py]: self.__dict__ is not {}

File: packit_service/service/events.py [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+    opened = "opened"
+    reopened = "reopened"
+    synchronize = "synchronize"
+
+
+class FedmsgTopic(enum.Enum):
+
+    receive = "org.fedoraproject.prod.git.receive"
+    copr_build_finished = "org.fedoraproject.prod.copr.build.end"
+    pagure_pr_added = "org.fedoraproject.prod.pagure.pull-request.flag.added"
```
Comment (TomasTomecek) [inline packit_service/service/events.py]: should be pr_flag_added

File: packit_service/service/events.py [hunk 5]
```diff
@@ -0,0 +1,206 @@
...
+class PullRequestAction(enum.Enum):
+
+    opened = "opened"
+    reopened = "reopened"
+    synchronize = "synchronize"
+
+
+class FedmsgTopic(enum.Enum):
+
+    receive = "org.fedoraproject.prod.git.receive"
```
Comment (TomasTomecek) [inline packit_service/service/events.py]: let's give this one a better name: how about dist_git_push? (that's what it is)

File: packit_service/service/events.py [hunk 6]
```diff
@@ -0,0 +1,206 @@
...
+    def get_project(self) -> GitProject:
+        config = Config.get_user_config()
+        pagure_service = PagureService(
+            token=config.pagure_user_token,
+            read_only=config.dry_run,
+            # TODO: how do we change to stg here? ideally in self.config
+        )
+        return pagure_service.get_project(
+            repo=self.repo_name, namespace=self.repo_namespace
+        )
```
Comment (TomasTomecek) [inline packit_service/service/events.py]: I really like how you broke the methods from steve into dedicated classes

File: packit_service/worker/parser.py [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+    InstallationEvent,
+    ReleaseEvent,
+    DistGitEvent,
+)
+from packit_service.worker.fedmsg_handlers import NewDistGitCommit
+
+logger = logging.getLogger(__name__)
+
+
+class Parser:
```
Comment (TomasTomecek) [inline packit_service/worker/parser.py]: docstring here would be nice, parser can really be anything

**Comment by TomasTomecek** [comment]
also needs a rebase now

File: packit_service/worker/github_handlers.py [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+                f"\nReturn code: {ex.rc}"
+            )
+            self.project.pr_comment(self.event.pr_id, msg)
+            msg = "Failed to create a SRPM."
+            r.report("failure", msg)
+            return HandlerResults(success=False, details={"msg": msg})
+        except FailedCreateSRPM:
+            msg = "Failed to create a SRPM."
+            r.report("failure", msg)
+            return HandlerResults(success=False, details={"msg": msg})
```
Comment (TomasTomecek) [inline packit_service/worker/github_handlers.py]: My fix is missing here: https://github.com/packit-service/packit-service/commit/9dbb371f1e6f3f72420ddea9f09f8c9a0655c47d

Comment (rpitonak) [reply packit_service/worker/github_handlers.py]: good catch! should be fixed now

**Comment by rpitonak** [comment]
The failing tests are unrelated, can we merge this? If there are no more changes requested.

 It is really painful to rebase this all the time 😄

**Pull request merged**